### ISSUE TITLE
✨ Add toggle to "Classification" to manage the filter on Search page

### ISSUE THREAD

**Issue by Jerryescandon** [GCTC-NTGC/gc-digital-talent#9483]
## ✨ Feature

Currently the classification filter on the search page is managed by hard-coding the available classifications into the dropdown list, we would like to create a functionality that allows admin users to control this directly through the admin page.

## 🕵️ Details

An admin should be able to:
1. Navigate to the classifications page located on the admin portal
2. Select a classification from the table
3. Edit the information from that classification
4. Turn the toggle ON or OFF
5. Have that change affect whether or not a classification shows up on the search page.

## 🎨 Design File

[Figma file](https://www.figma.com/file/Ov3ZwbSJXh7Ckqk6cefPDo/Admin---Classifications?type=design&node-id=5115%3A20828&mode=design&t=LWU99DTRUaNGgRuA-1) by @Jerryescandon 

### Target filter list located on search page

location: `https://talent.canada.ca/en/search`

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/16977254/273a3539-db14-403a-bb12-e41e1c19f586)

### New toggle to control visibility on the list

location: 

![Image](https://github.com/user-attachments/assets/2f7e2b39-b9dd-461e-a1bf-2ca341c7987a)

![Image](https://github.com/user-attachments/assets/59fe2849-a14a-466d-bcd8-728f16b1945f)

## 🙋‍♀️ Proposed Implementation

(optional) Do you have a proposed implementation?

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Platform Admins can control which classification-levels appear on the talent search page
- [ ] Search list uses the new display name field and fallback to regular name if empty
- [ ] Admins can assign a Display name for Group-level combinations that is displayed on the search page
- [ ] Classification admin table updated with new fields.
- [ ] Deployment instructions (seeder, artisan command, etc) to restore current list of classifications with their descriptions on the search page

**Comment by mnigh** [comment]
The open-ended nature of the **Display name** seems a bit dangerous for formatting and could lead to yet another source of truth for the "Generic Job Title", not to mention it does not provide for the aria-label value used in search (and, like the regular labels themselves, is hard coded). If this is intended to be the "Generic Job Title", why not call it that and use it elsewhere?

## Screenshot

<img width="1436" alt="Screen Shot 2024-04-18 at 12 33 54" src="https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/cea53c49-1b29-48ef-9e71-0686745d8ab3">

**Comment by gobyrne** [comment]
Good catch, we will need to make sure that all uses of these generic job titles are updated at the same time.

In terms of calling it "generic job title", I don't think we use that term on anything user-facing at this time. So best not to limit ourselves. It's worth considering though @Jerryescandon

**Comment by gobyrne** [comment]
Communities break-apart will affect this since there will be a new drop-down for selecting which community you request from. That choice should filter the classifications as well.

Will each community need their own toggle and Display name?

**Comment by vd1992** [comment]
@tristan-orourke 
Hard code for now
Discussion to decide how to best implement this

**Comment by gobyrne** [comment]
We decided to go forward with this. It's a bit awkward, but we want to move hardcoded stuff over to admin and this is top of the list.

**Comment by gobyrne** [comment]
@Jerryescandon still worth doing this in the context of your Admin work?

**Comment by esizer** [comment]
Any update on this? It has had the blocked label on it for some time now. Should it be moved back for more refining, iced or superseded by a new ticket?

**Comment by gobyrne** [comment]
I'm not sure why this is blocked, but there seems to be some questions about implementation. Let's punt it back to refinement, I've lowered the priority though.

**Comment by petertgiles** [comment]
Discussion in refinement:
- we will derive the "IT-01" part of the name so we can explode it for a11y
- this is different from generic job title: each model represents one group-level combination only and both generic job titles together

**Comment by petertgiles** [comment]
Asked Jerbo if we can remove the group-level prefix from display name

**Comment by petertgiles** [comment]
Team decided to re-review this at estimation due to discomfort with display name.
